lib/go/test/node_version_beacon_test.go

```diff
@@ -0,0 +1,198 @@
+	tx.SetScript([]byte(script))
+
+	tx.SetGasLimit(flowgo.DefaultMaxTransactionGasLimit)
+
+	err := tx.SignEnvelope(address, 0, signer)
+	if err != nil {
+		return err
+	}
+
+	err = b.AddTransaction(*tx)
+	if err != nil {
+		return err
+	}
+
+	_, _, err = b.ExecuteAndCommitBlock()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func TestNodeVersionBeacon(t *testing.T) {
+
+	b := newBlockchain()
+
+	env := templates.Environment{
+		ServiceAccountAddress: b.ServiceKey().Address.Hex(),
+	}
+
+	accountKeys := test.AccountKeyGenerator()
+
+	versionBeaconAccountKey, versionBeaconSigner := accountKeys.NewWithSigner()
+
+	versionBeaconContractScript := contracts.NodeVersionBeacon()
+
+	versionBeaconAddress, err := b.CreateAccount([]*flow.AccountKey{versionBeaconAccountKey}, nil)
+	assert.NoError(t, err)
+	_, err = b.CommitBlock()
+	assert.NoError(t, err)
+
+	env.NodeVersionBeaconAddress = versionBeaconAddress.String()
+
+	versionBeaconContract := sdktemplates.Contract{
+		Name:   "NodeVersionBeacon",
+		Source: string(versionBeaconContractScript),
+	}
+
+	versionThreshold := uint64(1234)
+
+	err = deployContract(b, versionBeaconAddress, versionBeaconSigner, versionBeaconContract, []cadence.Value{cadence.UInt64(versionThreshold)})
+	require.NoError(t, err)
+
+	t.Run("Should have initialized contract correctly", func(t *testing.T) {
+
+		result := executeScriptAndCheck(t, b, templates.GenerateGetVersionUpdateBufferScript(env), nil)
+
+		assertEqual(t, CadenceUInt64(versionThreshold), result)
+	})
+
+	t.Run("Should be able to send new version", func(t *testing.T) {
+
+		changeTx := createTxWithTemplateAndAuthorizer(b,
+			templates.GenerateAddVersionToTableScript(env),
+			versionBeaconAddress)
+
+		versionMajor := uint8(2)
+		err = changeTx.AddArgument(CadenceUInt8(versionMajor))
+		require.NoError(t, err)
+		versionMinor := uint8(13)
+		err = changeTx.AddArgument(CadenceUInt8(versionMinor))
+		require.NoError(t, err)
+		versionPatch := uint8(7)
+		err = changeTx.AddArgument(CadenceUInt8(versionPatch))
+		require.NoError(t, err)
+		preRelease := ""
+		err = changeTx.AddArgument(CadenceString(preRelease))
+		require.NoError(t, err)
+		isBackwardCompatible := true
+		err = changeTx.AddArgument(CadenceBool(isBackwardCompatible))
+		require.NoError(t, err)
+		versionHeight := versionThreshold + 44
+		err := changeTx.AddArgument(CadenceUInt64(versionHeight))
+		require.NoError(t, err)
+
+		txChangeResults := signAndSubmit(
+			t, b, changeTx,
+			[]flow.Address{versionBeaconAddress},
+			[]crypto.Signer{versionBeaconSigner},
+			false,
+		)
+		// no events just yet
+		assert.Len(t, txChangeResults.Events, 0)
+
+		checkTx := createTxWithTemplateAndAuthorizer(b,
+			templates.GenerateEmitVersionTableScript(env),
+			versionBeaconAddress)
+
+		txCheckResults := signAndSubmit(t, b, checkTx,
+			[]flow.Address{versionBeaconAddress},
+			[]crypto.Signer{versionBeaconSigner},
+			false,
+		)
+
+		assert.Len(t, txCheckResults.Events, 1)
```

Comment:
Would be good to add comments here and other similar places when you do an assert describing why the result should be what you're asserting

Reply:
I believe @m4ksio did those tests, I made a test case for this on the js tests.